Gems/AtomLyIntegration/AtomViewportDisplayInfo/Code/Source/AtomViewportDisplayInfoSystemComponent.cpp

```diff
@@ -163,8 +163,12 @@ namespace AZ::Render
         }
 
         m_drawParams.m_drawViewportId = viewportContext->GetId();
-        auto viewportSize = viewportContext->GetViewportSize();
-        m_drawParams.m_position = AZ::Vector3(static_cast<float>(viewportSize.m_width), 0.0f, 1.0f) + AZ::Vector3(r_topRightBorderPadding) * viewportContext->GetDpiScalingFactor();
+        
+        AZ::IConsole* console = AZ::Interface<AZ::IConsole>::Get();
+        uint32_t r_width;
+        console->GetCvarValue("r_width", r_width);
+
+        m_drawParams.m_position = AZ::Vector3(static_cast<float>(r_width), 0.0f, 1.0f) + AZ::Vector3(r_topRightBorderPadding) * viewportContext->GetDpiScalingFactor();
```

Comment:
Do you need to check if the r_resolutionMode is 1?
For the resolution mode which doesn't use r_width, this won't be correct.

Reply:
GetRenderResolution doesnt work as that will give the width after render scale is applied which will push the text to the middle of the screen. I do need r_width but for now I have wrapped it around r_resolutionMode. For the case where r_resolutionMode is 0 we probably need some code to handle Editor differently than launcher but that can come later. This fixes text rendering for mobile as r_resolutionMode is enabled for them. For platforms where you are running a gamelauncher and r_resolutionMode is not enabled will still have the issue where renderscale can move the positioning of debug text but that is of much less concern atm.